platforms/core-configuration/encryption-services/src/main/kotlin/org/gradle/internal/encryption/impl/DefaultEncryptionService.kt

```diff
@@ -65,7 +65,7 @@ class DefaultEncryptionService(
                     assertKeyLength(key)
                 }
             } catch (e: Exception) {
-                throw GradleException("Error loading encryption key from ${keySource.sourceDescription}", e)
+                throw EncryptionException("Error loading encryption key from ${keySource.sourceDescription}", e)
```

Comment:
ℹ️  Encryption service is initialized when `ConfigurationCacheStateStore` needs to be created. Previously it was done via `lazy { cacheRepository.forKey(cacheKey.string) }` in the `DefaultConfigurationCache`, which led to the exception thrown here, becoming the direct cause of the build failure.

As `ConfigurationCacheStateStore` is now consumed as a lazy service, the failure in its initialization (computing the `cacheKey.string`) becomes wrapped into the `ServiceCreationException`. Ultimately, this leads to the build failure report containing only this `Error loading encryption key ...` error, but not its cause. The reason for this behavior is that `GradleException` by itself is not `@Contextual`, and its causes are considered "interesting" for the user by the `BuildExceptionReporter`.

In order to keep the exception `e` caught here appearing on the console, the `GradleException` was replaced with a dedicated `EncryptionException` that is marked as `@Contextual`.

Reply:
For additional context, here is the difference in the output related to this exception wrapping:

Before:
```
* What went wrong:
Error loading encryption key from GRADLE_ENCRYPTION_KEY environment variable
> Encryption key length is 8 bytes, but must be at least 16 bytes long
```

After:
```
* What went wrong:
Gradle could not start your build.
> Could not create service of type ConfigurationCacheStateStore using ConfigurationCacheStoresProvider.createConfigurationCacheStateStore().
   > Error loading encryption key from GRADLE_ENCRYPTION_KEY environment variable
      > Encryption key length is 8 bytes, but must be at least 16 bytes long
```